packages/abstractions/types/uri-template-lite/index.d.ts

```diff
@@ -1,5 +1,3 @@
 declare module 'uri-template-lite' {
-  class Template {
-    constructor(template: string);
-  }
+  function expand(template: string, data: { [key: string]: unknown }): string;
```

Comment:
@andreaTP is there any specific reasons why we're not using the type definitions here?
https://www.npmjs.com/package/@types/uri-template-lite
Especially since the author of the type definitions is a Microsoft employee who works on Microsoft Graph and we can reach out if we need help.

Reply:
Besides the version comment, what makes you say they are outdated? especially in the context of using a single function.

For type definitions on DefinitelyTyped, you can just PR them, they have a bot that brings past contributors to review them. If you don't get traction let us know, I'll ping Vincenzo internally.

Reply:
as the user noticed this constructor is not working as expected after roll-up:

```ts
    class Template {
        constructor(template: string);
    }
```

probably because `Template` is a `function`: https://github.com/litejs/uri-template-lite/blob/654515a40bd0f01141119896f7f2b4e9cb23c106/index.js#L89

I'm not 100% sure how to properly fix the upstream types ...
I also think that the burden to keep this file in the repo is pretty low, and gives us shorter feedback loops when something changes.

Reply:
To be specific I'm not suggesting you change the code in this PR for RequestInformation. Just that you try to strip the types definitions from here to replace them with the @types package. The less code we have to maintain the better :)

Reply:
I completely understand, but I cannot find a working encoding of the exported `Template` ... it's a very very javascript thing ...

Reply:
wouldn't `import * as URI from 'uri-template-lite';` and then `URI.expand(this.urlTemplate, data);` work once this is removes and the types are added?

Reply:
Attempted a bunch of variations, in this case:

```
> @microsoft/kiota-abstractions@1.0.0-preview.19 build:cjs
> tsc -p tsconfig.cjs.json

src/requestInformation.ts:54:18 - error TS2339: Property 'expand' does not exist on type 'typeof import("/Users/aperuffo/workspace/kiota-typescript/node_modules/@types/uri-template-lite/index")'.

54       return URI.expand(this.urlTemplate, data);
                    ~~~~~~


Found 1 error in src/requestInformation.ts:54
```

Reply:
Note, in that case the following TS compiles:

```ts
URI.URI.expand(this.urlTemplate, data)
```

but it doesn't work at runtime in a vite build at least.

Reply:
Thanks a lot for the detailed research!
After this I'm more convinced that keeping this definition locally is the best course of action. :-) If you don't mind...